crates/store/README.md

```diff
@@ -1,163 +1,133 @@
+- [CheckNullifiersByPrefix](#checknullifiersbyprefix)
+- [GetAccountDetails](#getaccountdetails)
+- [GetAccountProofs](#getaccountproofs)
+- [GetAccountStateDelta](#getaccountstatedelta)
+- [GetBlockByNumber](#getblockbynumber)
+- [GetBlockHeaderByNumber](#getblockheaderbynumber)
+- [GetBlockInputs](#getblockinputs)
+- [GetNoteAuthenticationInfo](#getnoteauthenticationinfo)
+- [GetNotesById](#getnotesbyid)
+- [GetTransactionInputs](#gettransactioninputs)
+- [SyncNotes](#syncnotes)
+- [SyncState](#syncstate)
+<!--toc:end-->
+
+---
 
 ### ApplyBlock
 
 Applies changes of a new block to the DB and in-memory data structures.
 
-**Parameters**
+---
 
-- `block`: `BlockHeader` – block header ([src](../proto/proto/block_header.proto)).
-- `accounts`: `[AccountUpdate]` – a list of account updates.
-- `nullifiers`: `[Digest]` – a list of nullifier hashes.
-- `notes`: `[NoteCreated]` – a list of notes created.
+### CheckNullifiers
 
-**Returns**
+Gets a list of proofs for given nullifier hashes, each proof as a sparse Merkle Tree.
 
-This method doesn't return any data.
+---
 
-### CheckNullifiers
+### CheckNullifiersByPrefix
 
-Get a list of proofs for given nullifier hashes, each proof as a sparse Merkle Tree
+Returns a list of nullifiers that match the specified prefixes and are recorded in the node.
 
-**Parameters:**
+---
 
-- `nullifiers`: `[Digest]` – array of nullifier hashes.
+### GetAccountDetails
 
-**Returns:**
+Returns the latest state of an account with the specified ID.
 
-- `proofs`: `[NullifierProof]` – array of nullifier proofs, positions correspond to the ones in request.
+---
 
-### GetBlockHeaderByNumber
+### GetAccountProofs
 
-Retrieves block header by given block number. Optionally, it also returns the MMR path and current chain length to authenticate the block's inclusion.
+Returns the latest state proofs of the specified accounts.
 
-**Parameters**
+---
 
-- `block_num`: `uint32` _(optional)_ – the block number of the target block. If not provided, the latest known block will be returned.
+### GetAccountStateDelta
 
-**Returns:**
+Returns delta of the account states in the range from `from_block_num` (exclusive) to `to_block_num` (inclusive).
 
-- `block_header`: `BlockHeader` – block header.
+---
 
 ### GetBlockByNumber
 
 Retrieves block data by given block number.
 
-**Parameters**
+---
 
-- `block_num`: `uint32` – the block number of the target block.
+### GetBlockHeaderByNumber
 
-**Returns:**
+Retrieves block header by given block number. Optionally, it also returns the MMR path and current chain length to
+authenticate the block's inclusion.
 
-- `block`: `Block` – block data encoded using [winter_utils::Serializable](https://github.com/facebook/winterfell/blob/main/utils/core/src/serde/mod.rs#L26) implementation for [miden_objects::block::Block](https://github.com/0xPolygonMiden/miden-base/blob/main/objects/src/block/mod.rs#L43).
+---
 
 ### GetBlockInputs
 
 Returns data required to prove the next block.
```

Comment:
For this and similar endpoints, should we mention that they are meant to be used by the block producer?